Mage/src/main/java/mage/abilities/effects/common/CantCastEffect.java

```diff
@@ -0,0 +1,52 @@
+package mage.abilities.effects.common;
+
+import mage.MageObject;
+import mage.abilities.Ability;
+import mage.abilities.effects.ContinuousRuleModifyingEffectImpl;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.util.CardUtil;
+
+public class CantCastEffect extends ContinuousRuleModifyingEffectImpl {
```

Comment:
1. Must have additional param like `TargetController` (who can't cast), `Duration` (how long) or use correct class name like `OpponentsCantCastUntilYourNextTurnAllEffect` (AllEffect -- effect affects all cards, SourceEffect -- effect affects source card only, ControllerEffect -- effect affects controller's cards, etc).
2. Add usage comments what you need to use it in combination with `ChooseACardNameEffect`

Reply:
fixed -- made the name more specific and added comment that the effect should be used with `ChooseACardNameEffect`